### PR TITLE
Add n_images to DataLoader and some unit tests

### PR DESCRIPTION
**Pull request by moreshiny**
I put this together to avoid long test runtimes. Also adds a couple of tests. Submitting to devc as you may be working on this already @cverst Feel free to build on this or discard if you have something better. Will use this on devd in the meantime.

### CODE REVIEW

**Comment by cverst** [comment]
I updated the DataLoader differently and preserved more of the TensorFlow optimizations, resulting in less efficiency loss. I did use some code snippets of this pull request and will push a different version. The TestDataLoader can be merged afterwards.
